### DISCUSSION TITLE
How to modify button attribute with Pyscritp

### DISCUSSION THREAD

**Discussion by joseludenia** [pyscript/pyscript#1206] [Q&A] [answered]
Hello. I would like to know how a button can be enabled or disabled using pyscript. The code would look something like this, but using ·setAttribute" doesn't work.
Thank you so much

<div id="buttons">
<button  id="but" disabled>MYBUTTON</button>
 </div>

<py-script>
 def enabled():
        bot=Element("but")
        bot.setAttribute(disabled==False)
</py-script>

**Comment by FabioRosado** [comment]
Hello @joseludenia  you need to do `bot.elemnt.setAttribute`. Here's an example that should work:

```python
def enabled():
    button = Element("but").element.setAttribute("disabled", True)
```

**Reply by FabioRosado** [reply]
If you want to check if the button is already disabled you can do something like this:

```python
def enabled():
    button = Element("but").element
    disabled = button.disabled
    if not disabled:
        button.setAttribute("disabled", False)
```

**Reply by joseludenia** [reply]
Yes, that works, but when I try to activate a disabled button, like this, it doesn't work. Thank you so much.
Sorry, I don't know if you can see my code.

<button  id="but1" py-click="test()">button1</button>
<button  id="but2" disabled>button2</button>

<py-script>
def test():
	bot=Element("but2").element.setAttribute("disabled", False)
</py-script>

**Reply by FabioRosado** [reply]
Right that's because if a button is disabled event clicks won't register.

A workaround that is adding the button inside a div and if the button is disabled, clicking the div would set the button attribute to false.

Do you need help with an example code? I'm on the phone at the moment 😄

**Reply by FabioRosado** [reply]
Also are you disabling the button for style purposes? If so it might be worth trying to add/removing a specific class in the button that could mimics the disabled appearance.

**Reply by joseludenia** [reply]
Hello Fabio.
I can inactivate an activated button by clicking on another activate button, but curiously
I can not activate a disabled button by clicking another activate button. How can I do it?
Thank you so much.
Anoher thing, How can I insert code into this discussion like you have done before?

**Reply by FabioRosado** [reply]
You can insert code in two ways. Either by pressing the < > icon on the top right bar:

<img width="918" alt="Screenshot 2023-02-16 at 12 09 29" src="https://user-images.githubusercontent.com/3131401/219361206-3bb9d8a1-3407-4e92-9735-169f64a935e7.png">

You can also use triple backticks (`) if you can also specify the language by doing three backticks followed by the language such as Python.

**Comment by FabioRosado** [comment, marked as answer]
Here's a full working code that you can use:

```html
<!DOCTYPE html>
<html lang="en">
<head>
    <title>Enable Disable Buttons</title>
    <meta charset="UTF-8">
    <meta name="viewport" content="width=device-width,initial-scale=1">
    <link rel="stylesheet" href="https://pyscript.net/releases/2022.12.1/pyscript.css" />
    <script defer src="https://pyscript.net/releases/2022.12.1/pyscript.js"></script>
</head>
<body>
    <button id="button-1" py-click="toggle1()">Button 1</button>
    <button id="button-2" py-click="toggle2()">Button 2</button>
    <py-script >
        def toggle1():
            button1 = Element("button-1").element
            button2 = Element("button-2").element

            if button2.disabled:
                button2.removeAttribute("disabled")
            button1.setAttribute("disabled", True)

        def toggle2():
            button1 = Element("button-1").element
            button2 = Element("button-2").element

            if button1.disabled:
                button1.removeAttribute("disabled")
            button2.setAttribute("disabled", True)
    </py-script>

</body>
</html>
```

The code is a bit rough, but should give you an idea of how to do what you wish.  Also in py-script you can either write your python code between `<py-script></py-script>` tags like I did there. Or you can create a python file and point your py-script tags like this:

- Assume your code is in a file called `my-code.py` in the same directory as your `index.html`
- You can add `<py-script src="./my-code.py"></py-script>` and Pyscript will fetch your python code and run it in the same way as you would write in the `index.html` page.

This is useful if you want to use any help from your IDE (VsCode for example) such as code highlighting and autocompletion.

Hope this helps 😄

**Reply by joseludenia** [reply]
Yes!! It works!! Thank you so much.
Where did you find the "setAttribute" and "removeAttibute" attributes? These do not appear in the official Pyscript documentation.
Thanks again.

**Reply by FabioRosado** [reply]
These are DOM attributes specific for an element, that's why we are doing `Element("button-1").element` so we can access the attributes and change them. Here's the [documentation for setAttribute](https://developer.mozilla.org/en-US/docs/Web/API/Element/setAttribute) 😄
